.github/workflows/weekly.yml

```
@@ -0,0 +1,142 @@
+name: Weekly beta compile test
+
+on:
+  schedule:
+    # New versions of rust release on Thursdays. We test on Wednesdays to get at least a day's warning before all our CI breaks again.
+    # https://forge.rust-lang.org/release/process.html#release-day-thursday
+    - cron:  '0 12 * * 3'
+  workflow_dispatch:
+
+env:
+  CARGO_TERM_COLOR: always
+  NIGHTLY_TOOLCHAIN: nightly
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: [windows-latest, ubuntu-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@beta
+      - name: Install alsa and udev
+        run: sudo apt-get update; sudo apt-get install --no-install-recommends libasound2-dev libudev-dev
+        if: runner.os == 'linux'
+      - name: Build & run tests
+        # See tools/ci/src/main.rs for the commands this runs
+        run: cargo run -p ci -- test
+        env:
+          CARGO_INCREMENTAL: 0
+          RUSTFLAGS: "-C debuginfo=0 -D warnings"
+
+  ci:
```

```suggestion
  lint:
```

I think this could be made more specific, since this entire file is part of continuous integration.